### PR TITLE
Base for VFS Uplinks (FTP, SFTP)

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/main/java/sirius/biz/storage/layer3/uplink/util/RelativePath.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+package sirius.biz.storage.layer3.uplink.util;
+
+import sirius.biz.storage.layer3.MutableVirtualFile;
+import sirius.kernel.commons.Strings;
+
+/**
+ * Wraps a path to be attached to a {@link MutableVirtualFile} when providing an uplink.
+ */
+public class RelativePath {
```
Comment (jmuscireum) [inline src/main/java/sirius/biz/storage/layer3/uplink/util/RelativePath.java]: What is the benefit over using https://docs.oracle.com/javase/8/docs/api/java/io/File.html#File-java.lang.String-java.lang.String-

File: src/main/java/sirius/biz/storage/layer3/downlink/ssh/sftp/BridgeSeekableByteChannel.java [hunk 2]
```diff
@@ -21,9 +21,10 @@
 /**
  * Provides a simple implementation of the seekable channel.
  * <p>
- * This implementation doesn't actually support seeking, as this is not supported by the
- * {@link sirius.biz.storage.layer3.VirtualFileSystem}. Nevertheless, as long as all reads are sequential and don't
- * skip any data, we fullfill all calls.
+ * This implementation doesn't actually support much seeking, as this is not supported by the
+ * {@link sirius.biz.storage.layer3.VirtualFileSystem}. Nevertheless, as long as all writes are sequential and don't
+ * skip any data, we fullfill all calls. Also we can skip some bytes when reading as long as it is in the forward
```
Comment (idlira) [inline src/main/java/sirius/biz/storage/layer3/downlink/ssh/sftp/BridgeSeekableByteChannel.java]:
```suggestion
 * skip any data, we fulfill all calls. Also we can skip some bytes when reading as long as it is in the forward
```

File: src/main/java/sirius/biz/storage/layer3/uplink/util/RelativePath.java (hunk 1 above)
Comment (idlira) [reply src/main/java/sirius/biz/storage/layer3/uplink/util/RelativePath.java]: +1

File: src/main/java/sirius/biz/storage/layer3/uplink/util/UplinkConnectorPool.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        }
+    }
+
+    private GenericObjectPool<UplinkConnector<?>> createPool(UplinkConnectorConfig<?> uplinkConnectorConfig) {
+        UplinkConnectorFactory uplinkConnectorFactory = new UplinkConnectorFactory(uplinkConnectorConfig);
+        GenericObjectPool<UplinkConnector<?>> pool = new GenericObjectPool<>(uplinkConnectorFactory);
+        uplinkConnectorFactory.linkToPool(pool);
+        pool.setMaxIdle(uplinkConnectorConfig.maxIdle);
+        pool.setMaxTotal(uplinkConnectorConfig.maxActive);
+        pool.setTimeBetweenEvictionRunsMillis(30_000);
```
Comment (sabieber) [inline src/main/java/sirius/biz/storage/layer3/uplink/util/UplinkConnectorPool.java]: Extract into constants?

File: src/main/java/sirius/biz/storage/layer3/uplink/util/RelativePath.java (hunk 1 above)
Comment (andyHa) [reply src/main/java/sirius/biz/storage/layer3/uplink/util/RelativePath.java]: that this simply wraps a string describing a path on a remote server which doesn't share any api with a File in the local file system

Comment (idlira) [reply src/main/java/sirius/biz/storage/layer3/uplink/util/RelativePath.java]: As the JavaDoc states: An abstract representation of file and directory pathnames.
A java.io.File object does not necessarily means a real file in a local file system.

Comment (sabieber) [reply src/main/java/sirius/biz/storage/layer3/uplink/util/RelativePath.java]: Or a Path and its [resolve](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#resolve(java.nio.file.Path)) method could be used

**Pull request merged**